workflow-templates/im-test-cypress-schedule.yml

```
@@ -0,0 +1,184 @@
+  NPM_CACHE_PATH: ${{ github.workspace }}/node_modules # TODO: Update this to the path of the node_modules directory in the cypress test root directory
+
+jobs:
+  npm-cache:
+    runs-on: im-linux
+
+    defaults:
+      run:
+        shell: bash
+
+    outputs:
+      NPM_CACHE_KEY: ${{ env.NPM_CACHE_KEY }}
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.GITHUB_REF }}
+
+      - name: Set Cache Keys
+        # As long as these files don't change the cache will remain intact and the time
+        # to restore the different packages in the various jobs will be reduced.  If one
+        # of the files does change though, a new cache will be uploaded the next time
+        # it runs, then subsequent runs should be faster.
+        run: echo "NPM_CACHE_KEY=cypress_node_modules-${{ hashFiles('**/package-lock.json') }}" >> $GITHUB_ENV
+
+      - name: Check for an npm cache
+        id: has-cache
+        uses: actions/cache@v4
+        with:
+          lookup-only: true
+          key: ${{ env.NPM_CACHE_KEY }}
+          path: |
+            ${{ env.NPM_CACHE_PATH }}
+
+      # The remaining steps will only be executed if the cache was not found, otherwise they will be skipped.
+      # This action creates a post-job step that will upload the node_modules dir to the cache if the job
+      # completes successfully.  Subsequent jobs and workflow runs can use this cached version of the node_modules
+      # folder if the package-lock.json hasn't changed and it uses a ubuntu-20.04 runner to restore the cache from.
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Authenticate with GitHub Packages
+        uses: im-open/authenticate-with-gh-package-registries@v1.1
+        with:
+          read-pkg-token: ${{ secrets.READ_PKG_TOKEN }} # This is an org-level secret
+          orgs: 'im-enrollment,im-practices' # TODO: Verify list of orgs packages will be pulled from
+
+      - name: NPM Install
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        working-directory: '${{ env.CYPRESS_NPM_PACKAGE_PATH }}'
+        run: |
+          echo "Installing NPM"
+          npm ci
+
+      - name: NPM Install
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        working-directory: '${{ env.CYPRESS_NPM_PACKAGE_PATH }}'
+        run: |
+          echo "Installing NPM"
+          npm ci
+
+      - name: Save cache for node_modules directory
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v4
+        with:
+          key: ${{ env.NPM_CACHE_KEY }}
+          enableCrossOsArchive: true
+          path: |
+            ${{ env.NPM_CACHE_PATH }}
+
+  run-cypress:
+    runs-on: im-linux
```

```suggestion
    runs-on: im-linux
    needs: [npm-cache]
```

Need to add npm-cache as a pre-requisite.

---

Added.